### PR TITLE
updating label selector doc

### LINKED ISSUE #13699: update the description about label-selecors in the labels.md
The [documentation](https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/labels.md#label-selectors) suggests we support both "equality-based" and "set-based" label selectors, but actually we only support "==" selector right now. This is confusing. People are asking questions on IRC. We should update the doc in both github and k8s.io/v1.0.

### PR DESCRIPTION
**Pull request by sdminonne**
Attempt to fix #13699 
To clarify difference between `labels` `labels selector` and `set-based` vs `equalities-based` requirement 
@gmarek @caesarxuchao @bgrant0607

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra @kubernetes/rh-ux

File: docs/user-guide/labels.md [hunk 1]
```diff
@@ -121,15 +121,58 @@ _Set-based_ requirements can be mixed with _equality-based_ requirements. For ex
...
+$kubectl get pods -l 'environment in (production, qa)'
+```
+
+or restricting negative matching via _exists_ operator:
+
+```shell
+$kubectl get pods -l 'environment,environment notin (frontend)'
+```
+
+Kubernetes also supports two objects that use label selectors to keep track of their members, `service`s and `replicationcontroller`s:
```
Comment (thockin) [inline docs/user-guide/labels.md]: Not sure why we're documenting this - Service is not switching syntax is it?

Comment (sdminonne) [inline docs/user-guide/labels.md]: @thockin it's not that different of the doc we already have. See current doc at line 127

Comment (bgrant0607) [inline docs/user-guide/labels.md]: Per #13699, the main objective is to clarify the difference between the 2 different flavors of selectors.

**Pull request merged**